src/main/assets/js/document-manager.ts

```diff
@@ -0,0 +1,127 @@
+import Uppy from '@uppy/core';
+import DropTarget from '@uppy/drop-target';
+import FileInput from '@uppy/file-input';
+import ProgressBar from '@uppy/progress-bar';
+import XHRUpload from '@uppy/xhr-upload';
+
+import { DOCUMENT_MANAGER } from '../../steps/urls';
+
+import { getById } from './selectors';
+
+import '@uppy/drop-target/src/style.scss';
+import '@uppy/progress-bar/src/style.scss';
+
+const csrfToken = (getById('csrfToken') as HTMLInputElement)?.value;
+const endpoint = `${DOCUMENT_MANAGER}?_csrf=${csrfToken}`;
+
+const uploadProcessEl = getById('uploadProgressBar');
+const documentsInputEl = getById('uploadedDocuments') as HTMLInputElement;
+const noFilesUploadedEl = getById('noFilesUploaded');
+const filesUploadedEl = getById('filesUploaded');
+const errorUploadingEl = getById('uploadErrorSummary');
+const uploadGroupEl = getById('uploadGroup');
+
+location.hash = '';
+let files: { id: string; name: string }[] = JSON.parse(documentsInputEl?.value || '[]');
+const updateUploadedFiles = () => {
+  if (noFilesUploadedEl) {
+    if (files.length) {
+      noFilesUploadedEl.classList.add('govuk-visually-hidden');
+    } else {
+      noFilesUploadedEl.classList.remove('govuk-visually-hidden');
+    }
+  }
+
+  if (filesUploadedEl) {
+    filesUploadedEl.innerHTML = '';
+    for (const file of files) {
+      const filenameEl = document.createElement('li');
+      filenameEl.classList.add('uploadedFile');
+      filenameEl.textContent = file.name;
+
+      const deleteEl = document.createElement('a');
+      deleteEl.classList.add('govuk-link--no-visited-state');
+      deleteEl.href = `${DOCUMENT_MANAGER}/delete/${file.id}`;
+      deleteEl.textContent = 'Delete';
+      deleteEl.onclick = async e => {
+        e.preventDefault();
+        (e.target as HTMLAnchorElement).style.cursor = 'wait';
+        document.body.style.cursor = 'wait';
+
+        try {
+          const request = await fetch(endpoint, {
+            method: 'DELETE',
```

Comment:
I'm confused by all of this... why are we deleting things?

Reply:
This is part of [NFDIV-640](https://tools.hmcts.net/jira/browse/NFDIV-640) Delete uploaded documents